modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/LambdaMultiplePropertyChangeListenerHandler.java

```diff
@@ -158,6 +158,32 @@ public final void registerInvalidationListener(Observable observable, Consumer<O
         observableReferenceMap.merge(observable, consumer, Consumer::andThen);
     }
 
+    /**
+     * Executes a lambda if the property value is not null, otherwise adds a listener to the ObservableValue {@code p}
+     * to wait until it is set and calls the lambda then, followed by removing the said listener.
+     *
+     * @param p Observable value
+     * @param consumer lambda to invoke when the property value is not null
+     */
+    public <T> void executeOnceWhenPropertyIsNonNull(ObservableValue<T> p, Consumer<T> consumer) {
+        if (p == null) {
+            return;
+        }
+
+        T value = p.getValue();
+        if (value != null) {
+            consumer.accept(value);
+        } else {
+            registerInvalidationListener(p, (x) -> {
+                T v = p.getValue();
+                if (v != null) {
+                    unregisterInvalidationListeners(p);
+                    consumer.accept(v);
+                }
+            });
```

Comment:
beware: unregister removes _all_ listeners (of type InvalidationListener) from the given property

Reply:
this makes no sense!  thank you for pointing it out.
perhaps we really *ought to* create a better LambdaMultiplePropertyChangeListenerHandler implementation.

Reply:
well, it _does_ make sense for listeners registered on properties of the control itself (they don't change and are un/registered once) - and that was it was designed for :) What it can't handle (and was not designed for) are dynamic multiple un/registration of listeners.

Reply:
btw: the executeOnce pattern is appropriate only if it's guaranteed that the value changes once from null to not-null (not back again) .. which is not the case for the value of a control's scene property - it can change in both directions or not at all (which might be the reason for having added a second listener ;)

Reply:
you are so right here.